ergo/distributions/histogram.py

```diff
@@ -104,8 +104,54 @@ def sample(self):
     def rv(self):
         raise NotImplementedError
 
-    def normalize(self):
-        return HistogramDist(self.logps, 0, 1)
+    def normalize(self, true_scale: scale.Scale = None):
+        """
+        Normalize the histogram onto [0,1]
+
+        Setting a true_scale allows you to express that the histogram to be normalized
+        did not cover the entire scale of interest.
+
+        E.g. -- imagine my histogram only has probability mass from 2 to 5,
+        but I'm interested in p over [0,10].
+        So I'll use Scale(0,10), and I'll get back a histogram with additional
+        bins with 0 probability to cover the area from 0 to 2 and 5 to 10,
+        where my histogram doesn't place any probability.
+
+        :param true_scale: the full scale that I'm interested in probability over.
+        """
+        # if true_scale is not provided, assume that the histogram has
+        # entries over the entire scale of interest
+        if not true_scale:
+            return HistogramDist(self.logps, 0, 1)
+
+        if true_scale.scale_min is None or true_scale.scale_max is None:
+            raise ValueError(
+                "If you provide a true_scale, you must provide both a min and max"
+            )
+
+        if (
+            true_scale.scale_min > self.scale_min
+            or true_scale.scale_max < self.scale_max
+        ):
+            raise ValueError(
+                "Can only rescale hist to a scale that includes all of its current scale"
+            )
+
+        x_range_below = scale.Scale(true_scale.scale_min, self.scale_min)
+        x_range_below_per_hist_range = x_range_below.range / self.scale.range
+        num_x_bins_below = round(self.size * x_range_below_per_hist_range)
```

Comment:
Note that this rounding introduces some imprecision. But it seems hard to avoid without one of the refactors mentioned above